wrappers/python/openmm/app/simulation.py

```diff
@@ -226,19 +232,33 @@ def _simulate(self, endStep=None, endTime=None):
                 unwrapped = []
                 either = []
                 for reporter, report in zip(self.reporters, nextReport):
-                    if report[0] == nextSteps:
-                        if len(report) > 5:
-                            wantWrap = report[5]
-                            if wantWrap is None:
+                    if isinstance(report, dict):
+                        if report['steps'] == nextSteps:
+                            wantWrap = self._usesPBC if report['periodic'] is None else report['periodic']
```

Comment:
This could have unintended side effects. Currently, some reporters use None as value to get the default behavior (e.g. the [dcdreporter](https://github.com/openmm/openmm/blob/f48eb17ba3a537e93beb1a60bb3e8c7b4099d3dc/wrappers/python/openmm/app/dcdreporter.py)).

Changing it in the included reporters should not be an issue, but it could change how custom reporters behave.

I also think we should make it possible to explicitly request to use the system default.

I would propose the following:

`wantWrap = self._usesPBC if report.get('periodic') is None else report['periodic']`

That would catch all possibilities and return the default value in both cases, when report['periodic'] is undefined and when it is set to `None`, explicitly requesting the sytem default.